main.go

```diff
@@ -98,27 +96,32 @@ func main() {
 		panic(err)
 	}
 
-	// Parse File
-	var config map[string][]string
-	if err := yaml.Unmarshal(yamlFile, &config); err != nil {
+	// Parse Terrafile
+	sourceDependenciesMap, err := parseTerrafile(yamlFile)
+	if err != nil {
+		panic(err)
+	}
+
+	// Cleanup module path
+	if err := os.RemoveAll(opts.ModulePath); err != nil {
 		panic(err)
 	}
 
-	// Clone modules
-	os.RemoveAll(opts.ModulePath)
-	os.MkdirAll(opts.ModulePath, os.ModePerm)
-	for source, refs := range config {
+	if err := os.MkdirAll(opts.ModulePath, os.ModePerm); err != nil {
+		panic(err)
+	}
+
+	for source, dependencies := range sourceDependenciesMap {
```

Comment:
Just to clarify, if we're using a segment terrafile here, would this map look like this example?
```
{
  terracode-modules: []Dependency{
    { Alias: nil, SourceRef: &SourceRef{ Source: "terracode-modules", Version: "chamber_v2.0.0"} },
    { Alias: nil, SourceRef: &SourceRef{ Source: "terracode-modules", Version: "iam_v1.0.0"} },
  }
}
```

I just want to make sure we're preserving the behaviour of cloning once and doing a checkout for every version. This was one of the changes we made from upstream to improve performance, since they would do a checkout for every source. In the case of terracode-modules, this was a big performance hit since it's quite a large repo.

Reply:
I think this performance hit will disappear once we can implement a global module cache on the Atlantis host. Right now these terrafile modules are cached per pull request, so we have to re-download the modules every commit.

Reply:
Yes, your example is correct

I grouped it by source on purpose, so we can preserve behavior of cloning once and doing checkout of every version.